### PR TITLE
Bump Workbench to v2024.09.0

### PR DESCRIPTION
**Pull request by zachhannum**

### CODE REVIEW

**Review by melissa-barca — APPROVED** [review]
LGTM!

**Comment by bschwedler** [comment]
@zachhannum Could you please update the goss tests for [GCW](https://github.com/rstudio/rstudio-docker-products/blob/fae3b60e30dd4f3da596e571e130bbd3bfdd28fe/workbench-for-google-cloud-workstations/test/goss.yaml#L73-L74) and [azureML](https://github.com/rstudio/rstudio-docker-products/blob/fae3b60e30dd4f3da596e571e130bbd3bfdd28fe/workbench-for-microsoft-azure-ml/test/goss.yaml#L81-L82) to use the updated code-server binary location?

We can probably remove the [conditional logic](https://github.com/rstudio/rstudio-docker-products/blob/fae3b60e30dd4f3da596e571e130bbd3bfdd28fe/workbench/test/goss.yaml#L73-L79) in the main workbench gos tests.

```log
Failures/Skipped:

File: /usr/lib/rstudio-server/bin/code-server/bin/code-server: exists:
Expected
    false
to equal
    true

Total Duration: 20.209s
Count: 234, Failed: 1, Skipped: 0
```

**Review by melissa-barca — CHANGES_REQUESTED** [review]
Oops was too quick with my LGTM, we'll need to make some changes to the vscode.conf files to use PWB Code Server and possibly other updates as well. I'll push a commit for this.

**Comment by bschwedler** [comment]
@melissa-barca in parallel, I am updating the R/Python versions for Workbench to the latest patch release in #846

**Review by melissa-barca — APPROVED** [review]

File: workbench-for-google-cloud-workstations/conf/vscode.conf [hunk 1]
```diff
@@ -1,4 +1,4 @@
 enabled=1
-args=--verbose --host=0.0.0.0
+args=--host=0.0.0.0
```
Comment (GCRev) [inline workbench-for-google-cloud-workstations/conf/vscode.conf:2]: 👍

**Review by GCRev — APPROVED** [review]
LGTM

**Comment by bschwedler** [comment]
I changed the merge target to the `dev` branch so that we can release the R/Python updates and the workbench updates at the same time.

**Review by ianpittwood — APPROVED** [review]

**Review by bschwedler — APPROVED** [review]

**Pull request merged**